src/main/clm_driver.F90

```diff
@@ -1226,23 +1226,22 @@ subroutine clm_drv(doalb, nextsw_cday, declinp1, declin, rstwr, nlend, rdate, ro
        ! ============================================================================
        ! Determine albedos for next time step
        ! ============================================================================
-       
-       if (.not.doalb) then
-
 
-          if(use_fates) then
-             ! During branch runs and non continue_run restarts, the doalb flag
-             ! does not trigger correctly for fates runs (and non-fates?), and thus
-             ! the zenith angles are not calculated and ready when radiation scattering
-             ! needs to occur.
-             call UpdateZenithAngles(bounds_clump, surfalb_inst, nextsw_cday, declinp1)
-             call clm_fates%wrap_canopy_radiation(bounds_clump, nc, &
+       ! This is only relevant to  fates two stream to not break sun fraction calculations
+       ! on the second timestep after start from finidat or for hybrid run.
+       if (use_fates .and. .not. use_fates_sp .and. fates_radiation_model == 'twostream') then
```

Comment:
@mvdebolskiy , why did you add the .not.use_fates_sp condition here?

Reply:
I don't remember exactly, actually. Maybe the variables that don't get initialized in nocomp are actually ok in sp.